pkg/bff/api/v1/errors.go

```diff
@@ -0,0 +1,52 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+var (
+	unsuccessful = Reply{Success: false}
+	notFound     = Reply{Success: false, Error: "resource not found"}
+	notAllowed   = Reply{Success: false, Error: "method not allowed"}
+)
+
+// ErrorResponse constructs an new response from the error or returns a success: false.
+func ErrorResponse(err interface{}) Reply {
+	if err == nil {
+		return unsuccessful
```

Comment:
What are some ways we think `unsuccessful` might come up?

Reply:
I remember this being added to prevent a panic in the Admin API when we were accidentally passing `nil` into `ErrorResponse` (I believe nil was being sent on an error channel, which is what caused the confusion). 

In terms of the use case, I could imagine this response being sent when the error code matters more, for example, `c.JSON(http.StatusPayloadTooLarge, ErrorResponse(nil))` -- here, it's largely the client's responsibility to break the payload into chunks using a multipart upload. The JSON body doesn't matter. However, even in this case, I'd prefer `c.JSON(http.StatusPayLoadTooLarge, ErrorResponse("payload is too large, upload in chunks"))` -- so mostly it is just panic prevention.